source/dofs/dof_handler.cc

```diff
@@ -1293,8 +1293,8 @@ DoFHandler<dim, spacedim>::distribute_mg_dofs()
   // only if this is a sequential
   // triangulation. it doesn't work
   // correctly yet if it is parallel
-  if (dynamic_cast<const parallel::distributed::Triangulation<dim, spacedim> *>(
-        &*tria) == nullptr)
+  if (dynamic_cast<const parallel::DistributedTriangulationBase<dim, spacedim>
+                     *>(&*tria) == nullptr)
```

Comment:
If you want, put this into a PR of its own. It's correct even without the rest of your patch.

Reply:
Done! See PR #8724.